[JSC] Stop ignoring useWebAssembly=1 when JIT is disabled

rkirsling opened the pull request:
#### 2c0494fcc3fe018670c408476931c3698d9b2ee2
<pre>
[JSC] Stop ignoring useWebAssembly=1 when JIT is disabled
<a href="https://bugs.webkit.org/show_bug.cgi?id=263763">https://bugs.webkit.org/show_bug.cgi?id=263763</a>

Reviewed by Justin Michaud.

It&apos;s true that WebAssembly won&apos;t yet *run* sensibly with JIT off,
but ENABLE(WEBASSEMBLY) &amp;&amp; !ENABLE(JIT) *is* a valid build configuration now,
so JSC should permit enabling useWebAssembly even when JIT is disabled.

* Source/JavaScriptCore/runtime/Options.cpp:
(JSC::disableAllJITOptions):

Canonical link: <a href="https://commits.webkit.org/270100@main">https://commits.webkit.org/270100@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b3e5248e7913286e5961ec37da66682f355a6a94

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/24231 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/2340 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25313 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26363 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22306 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/4001 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24698 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22756 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24474 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1885 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20940 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26952 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1630 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21861 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/28078 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/21073 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/22113 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/22160 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25871 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/23529 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1539 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/19198 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/30927 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/2663 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/6787 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5885 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1945 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/30891 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1910 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/6462 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

justinmichaud commented on JSTests/wasm/noJIT/noJIT.js:
```diff
@@ -1,2 +0,0 @@
-if (typeof WebAssembly !== "undefined")
```
We should make wasm be off by default if you don't have jit support for now (which should let this test pass), and then when we can eventually actually run wasm, we should run this test with --useWebAssembly=0

rkirsling replied:
Alrighty, redid the PR accordingly.

justinmichaud reviewed (approved):
r=me

rkirsling committed:
[JSC] Stop ignoring useWebAssembly=1 when JIT is disabled
https://bugs.webkit.org/show_bug.cgi?id=263763

Reviewed by Justin Michaud.

It's true that WebAssembly won't yet *run* sensibly with JIT off,
but ENABLE(WEBASSEMBLY) && !ENABLE(JIT) *is* a valid build configuration now,
so JSC should permit enabling useWebAssembly even when JIT is disabled.

* Source/JavaScriptCore/runtime/Options.cpp:
(JSC::disableAllJITOptions):

Canonical link: https://commits.webkit.org/270100@main
